src/include/ipc/probe.h

```diff
@@ -0,0 +1,183 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2019 Intel Corporation. All rights reserved.
+ *
+ * Author: Adrian Bonislawski <adrian.bonislawski@linux.intel.com>
+ *         Artur Kloniecki <arturx.kloniecki@linux.intel.com>
+ */
+
+/**
+ * \file include/ipc/probe.h
+ * \brief Probe IPC definitions
+ * \author Adrian Bonislawski <adrian.bonislawski@linux.intel.com>
+ * \author Artur Kloniecki <arturx.kloniecki@linux.intel.com>
+ */
+
+#ifndef __IPC_PROBE_H__
+#define __IPC_PROBE_H__
+
+#include <ipc/header.h>
+#include <sof/bit.h>
+#include <stdint.h>
+
+#define PROBE_PURPOSE_EXTRACTION	0x1
+#define PROBE_PURPOSE_INJECTION		0x2
+
+#define PROBE_EXTRACT_SYNC_WORD		0xBABEBEBA
+
+/**
+ * \brief Definitions of shifts and masks for format encoding in probe
+ *	  extraction stream
+ *
+ * Audio format from extraction probes is encoded as 32 bit value. Following
+ * graphic explains encoding.
+ *
+ * A|BBBB|CCCC|DDDD|EEEEE|FF|GG|H|I|J|XXXXXXX
+ * A - 1 bit - Specifies Type Encoding - 1 for Standard encoding
+ * B - 4 bits - Specify Standard Type - 0 for Audio, 1 for Video
+ * C - 4 bits - Specify Audio format - 0 for PCM, 1 for MP3
```

Comment:
I missed this early, did anyone suggest that the data would be re-encoded in MP3? What is the expected usage here? If it's not clear maybe we should just list 0-PCM

Reply:
There would be no re-encoding, absolutely. But the interface inherited from cAVS and documented in FDK specifies what possible values might get there. Same with video on the line above. I might remove these, as they would not occur in our data packets in any foreseeable future.

Reply:
ok, let's remove.

Reply:
obsolete documentation removed